File: 2015/day02-1.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func handle_error(err error) {
	if err != nil {
		fmt.Println(err)
		return
	}
}

func main() {
	content, err := os.ReadFile("./data/input2.txt")
	handle_error(err)

	lines := strings.Split(string(content), "\n")
	total := 0
	for _, line := range lines[0 : len(lines)-1] {
		sides := make([]int, 3)
		parts := strings.Split(line, "x")
		length, err := strconv.Atoi(parts[0])
		handle_error(err)
		width, err := strconv.Atoi(parts[1])
		handle_error(err)
		height, err := strconv.Atoi(parts[2])
		handle_error(err)

		sides[0] = length * width
		sides[1] = width * height
		sides[2] = length * height

		min := sides[0]
		for _, side := range sides {
			if side < min {
				min = side
			}
		}
		paper_needed := 2*sides[0] + 2*sides[1] + 2*sides[2] + min

		total += paper_needed
	}
	fmt.Println(total)
}

File: 2017/day02-1.js
import fs from 'fs'

const data = fs.readFileSync('./data/input2.txt', 'utf-8')

const lines = data.split('\n')
lines.pop()

let checksum = 0
for (const line of lines) {
  const nums = line.split('\t').map(x => parseInt(x))
  const min = Math.min(...nums)
  const max = Math.max(...nums)
  const diff = max - min
  console.log(min, max, diff)
  checksum += diff
}

console.log(checksum)

